src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,250 @@
+
+        Assertions.assertEquals(actual, expected);
+        Assertions.assertFalse(Storage.people.contains(fourthUser),
+                "User should be added to storage.");
+    }
+
+    @Test
+    void userPasswordLength_isOk() {
+        var actual = registrationService.register(firstUser).getLogin().length();
+
+        Assertions.assertTrue(MIN_LENGTH <= actual);
+        Assertions.assertTrue(Storage.people.contains(firstUser),
+                "User should be added to storage.");
+    }
+
+    @Test
+    void userPasswordLength_isNotOk() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(fifthUser));
+
+        var expected = "Not valid password."
+                + "Min length is "
+                + MIN_LENGTH;
+        var actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertFalse(Storage.people.contains(fifthUser),
+                "User with short password should not be added to storage.");
+    }
+
+    @Test
+    void userPassword_isNull() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(sixthUser));
+
+        var expected = "User password cannot be null";
+        var actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertFalse(Storage.people.contains(sixthUser),
+                "User with null password should not be added to storage.");
+    }
+
+    @Test
+    void userAges_isOk() {
+        var actual = registrationService.register(firstUser).getAge();
+
+        Assertions.assertTrue(MIN_AGE <= actual);
+        Assertions.assertTrue(Storage.people.contains(firstUser),
+                "User should be added to storage.");
+    }
+
+    @Test
+    void userAge_isNotOk() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(seventhUser));
+
+        var expected = "Not valid age "
+                + seventhUser.getAge()
+                + ". Min age is " + MIN_AGE;
+        var actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertFalse(seventhUser.getAge() > MIN_AGE);
+        Assertions.assertFalse(Storage.people.contains(seventhUser),
+                "User with incorrect age should not be added to storage.");
+    }
+
+    @Test
+    void userAge_isNegative() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(eighthUser));
+
+        var expected = "Not valid age "
+                + eighthUser.getAge()
+                + ". Because age is negative";
+        var actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertFalse(eighthUser.getAge() > MIN_AGE);
```

Comment:
The assertion `Assertions.assertFalse(eighthUser.getAge() > MIN_AGE);` is incorrect. It should check that the user's age is not greater than or equal to `MIN_AGE`, or simply be removed since the age validation is already covered by the exception.